cpp/oneapi/dal/algo/svm/backend/gpu/test/working_set.cpp

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/io.hpp"
+
+#include "oneapi/dal/algo/svm/backend/gpu/working_set.hpp"
+
+namespace oneapi::dal::svm::backend::test {
+
+namespace pr = dal::backend::primitives;
+namespace te = dal::test::engine;
+namespace la = te::linalg;
+
+template <typename TestType>
+class working_set_test : public te::policy_fixture {
+public:
+    using Float = TestType;
+
+    void test_working_set(const std::vector<Float>& f,
```

Comment:
```c++
ndarray<Float, 1> get_smoke_f() {
   const std::int64_t count = 10
   const float_t f_host[] = { 2, 7, 3, 9, 5, 1, 8, 4, 6 };
   auto f = ndarray<Float, 1>::empty(q, { count }, sycl::usm::alloc::device);
   f.assign(q, f_host, count).wait_and_throw();
   return f;
}
```

Reply:
@rlnx, I have different input arrays, so it won't work to make one common function to get ndarray of f / y / alpha